magda-indexer/src/main/scala/au/csiro/data61/magda/indexer/search/elasticsearch/ElasticSearchIndexer.scala

```diff
@@ -375,27 +369,9 @@ class ElasticSearchIndexer(
     }
   }
 
-  override def index(dataSetStream: Source[DataSet, NotUsed]) = {
-    val indexResults = dataSetStream
-      // Queue every dataSet for indexing and keep the future along with the dataset's identifier
-      .map(dataSet => (dataSet.uniqueId, index(dataSet)))
-      // Combine all of these operations into one future
-      .runWith(Sink.fold(Future(new SearchIndexer.IndexResult(0, Seq()))) {
-        case (combinedResultFuture, (thisResultIdentifier, thisResultFuture)) =>
-          combinedResultFuture.flatMap { combinedResult =>
-            thisResultFuture.map { _ =>
-              combinedResult.copy(successes = combinedResult.successes + 1)
-            }.recover {
-              case (e: Throwable) =>
-                combinedResult.copy(failures = combinedResult.failures :+ thisResultIdentifier)
-            }
-          }
-      })
+  // TODO: Instead of using indexQueue that performs bulk indexing, make single indexing request.
+  private def index2(dataSet: DataSet, promise: Promise[Unit] = Promise[Unit]): Future[Unit] = {
```

Comment:
Can we name this something more descriptive? What does it do that's different from `index` ?

Reply:
The original implementation uses the name "index", the same as another "index" that is api. I only renamed it to "index2". I can revert it back if you prefer. As described in the TODO comment, this part will be re-written. I have not figured out how to do it yet.